fuel-vm/src/interpreter/ecal.rs

```diff
@@ -0,0 +1,93 @@
+use fuel_asm::RegId;
+use fuel_tx::DependentCost;
+use fuel_types::Word;
+
+use crate::{
+    call::CallFrame,
+    error::SimpleResult,
+};
+
+use super::Interpreter;
+
+pub trait EcalAccess {
```

Comment:
If anyone can figure out a better way than having this trait, I'd be delighted to hear about it

Reply:
I would prefer more the approach with `Ecall`/`External` generic in `Interpreter`. It is more perform, plus we don't need to define the trait with a limited number of functions that should be implemented by the `Interpreter`. Instead, the developer can have access to all public functions of the `Interpreter`.

[Here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1a86cb5176f88c1f46ad53bac271ec3f) is an example of how it can work.

Reply:
Done in 51150738. I feel like it's super ugly as it pollutes all `impl` singatures of `Interpreter` and `Transactor`. Default value doesn't help us much.